### PR TITLE
passport auth for profiles and companies routes

### PR DESCRIPTION
**Pull request by NikolaGolubovic**

### CODE REVIEW

**Comment by NikolaGolubovic** [comment]
Somehow I overlooked your directions @vuxor  to change requests for /register route... Will change soon...

File: package.json [hunk 1]
```diff
@@ -44,8 +46,12 @@
     "cors": "^2.8.5",
     "dotenv": "^10.0.0",
     "express": "^4.17.2",
+    "express-session": "^1.17.2",
```
Comment (vuxor) [inline package.json]: Do you actually need this package?

File: src/models/users.ts [hunk 2]
```diff
@@ -2,11 +2,11 @@ import { DataTypes, Model } from "sequelize";
...
-  public userId!: number;
-  public username!: string;
-  public email!: string;
-  public password!: string;
-  public role!: "company-user" | "company-admin" | "superadmin";
+  declare userId: number;
+  declare username: string;
+  declare email: string;
+  declare password: string;
+  declare role: "company-user" | "company-admin" | "superadmin";
```
Comment (vuxor) [inline src/models/users.ts]: Role values should be saved and used as variables

File: src/services/authService.ts [hunk 3]
```diff
@@ -3,10 +3,23 @@ import jwt from "jsonwebtoken";
...
   try {
     const { username, password: rowPass, email } = parseRegisterBody(req);
+    const {
+      companyName,
+      companyLogo,
+      profileName,
+      profilePhoto,
+      profileStatus,
+    } = req.body;
+    console.log(
```
Comment (vuxor) [inline src/services/authService.ts]: Remove console.log

File: src/services/authService.ts [hunk 4]
```diff
@@ -21,20 +34,68 @@ export const registerUser: RequestHandler = async (req, res, next) => {
...
     });
-    res.status(200).json({ user: newUser });
+    const newUserFound = await User.findOne({ where: { username: username } });
+    if (newUserFound === null) {
+      await newUserFound.destroy();
+      throw new Error("Something went wrong with new user creation");
+    }
+
+    await Company.create({
+      name: companyName || `${username}'s Country`,
```
Comment (vuxor) [inline src/services/authService.ts]: Replace country with company

File: src/services/authService.ts [hunk 5]
```diff
@@ -21,20 +34,68 @@ export const registerUser: RequestHandler = async (req, res, next) => {
...
+      logo: companyLogo || `${username}-logo.jpg`,
+      slug: companyName
+        ? slugify(companyName, { lower: true })
+        : `${slugify(username.toLowerCase())}-country-company`,
```
Comment (vuxor) [inline src/services/authService.ts]: Remove country from slug

File: src/services/authService.ts [hunk 6]
```diff
@@ -21,20 +34,68 @@ export const registerUser: RequestHandler = async (req, res, next) => {
...
+      updatedAt: new Date(),
+      userId: +newUserFound.userId,
+    });
+    console.log("newFoundUserID", newUserFound.userId);
+    const newCompanyFound = await Company.findOne({
+      where: { userId: newUserFound.userId },
+    });
+    console.log("newCompanyFound", newCompanyFound);
```
Comment (vuxor) [inline src/services/authService.ts]: Remove console logs

File: src/services/authService.ts [hunk 7]
```diff
@@ -21,20 +34,68 @@ export const registerUser: RequestHandler = async (req, res, next) => {
...
+    if (newCompanyFound === null) {
+      await newUserFound.destroy();
+      await newCompanyFound.destroy();
```
Comment (vuxor) [inline src/services/authService.ts]: If new company is null, then you will not be able to call destroy method

File: src/services/authService.ts [hunk 8]
```diff
@@ -21,20 +34,68 @@ export const registerUser: RequestHandler = async (req, res, next) => {
...
+      companyId: +newCompanyFound.companyId,
+    });
+    const newProfileFound = await Profile.findOne({
+      where: {
+        userId: newUserFound.userId,
+        companyId: newCompanyFound.companyId,
+      },
+    });
+    if (newProfileFound === null) {
+      await newProfileFound.destroy();
```
Comment (vuxor) [inline src/services/authService.ts]: Same as for new company

File: src/models/users.ts (hunk 2 above)
Comment (NikolaGolubovic) [reply src/models/users.ts]: sorry, this suggestion was too unspecific for me so i simply created typescript enum solution for these three

**Review by vuxor — APPROVED** [review]

**Pull request merged**